path_role.go

```diff
@@ -41,6 +41,10 @@ are allowed, both this and bound_service_account_namespaces can not be "*"`,
 					Description: `List of namespaces allowed to access this role. If set to "*" all namespaces
 are allowed, both this and bound_service_account_names can not be set to "*"`,
 				},
+				"audience": &framework.FieldSchema{
```

Comment:
Per my comment at https://github.com/terraform-providers/terraform-provider-vault/pull/601#discussion_r353856689, did you consider accepting a set of audiences here, requiring nonempty intersection with the audiences claimed by the token?